moveit_core/transforms/src/transforms.cpp

```diff
@@ -123,7 +123,7 @@ void Transforms::setTransform(const geometry_msgs::TransformStamped& transform)
 {
   if (sameFrame(transform.child_frame_id, target_frame_))
   {
-    Eigen::Affine3d t = tf2::transformToEigen(transform.transform);
+    Eigen::Isometry3d t = tf2::transformToEigen(transform.transform);
```

Comment:
```suggestion
    Eigen::Isometry3d t = Eigen::Isometry3d(tf2::transformToEigen(transform.transform).matrix());
```
it ain't pretty but ...

Reply:
Did you tested this? I don't think it will work. Affine is more general than Isometry and thus cannot be converted.

Reply:
I tested that the following compiles:


    Eigen::Isometry3d test;

    Eigen::Affine3d test2(test);

    Eigen::Isometry3d test3(test2.matrix());

    Eigen::Isometry3d test4 = Eigen::Isometry3d(test2.matrix());


 I think that `.matrix()` is the equivalent of `std::string::c_str()`, ie go to raw data and let the user ensure type safety etc.

Reply:
In best case, it does would it should do. In worst case, Eigen will forbid this kind of conversion later on. In any case it's ugly. I'd prefer to file a PR for `tf2_eigen` to deal with this, which should be done in any case, as `tf2_eigen` uses `rotation()` as well, which is computing a costly SVD.

Reply:
This also requires copying the matrix. Not as costly as SVD, but not necessary either.

Reply:
> Eigen will forbid this kind of conversion in future

That fear seems unfounded - what makes you think they will change this conversion API?

I like that @simonschmeisser's approach works for now, even if we get an upstream change in eventually